src/port.js

```diff
@@ -114,6 +141,23 @@ export default class Port {
 		}
 
 	}
+	registerService(serviceType, version, service) {
+		if(this.isConnected) {
+			throw new Error('Register services before connecting');
+		}
+		if(!typeNameValidator.test(serviceType)) {
+			throw new Error(`Invalid service type "${serviceType}"`);
+		}
+		let methodNames = [];
+		for(let p in service) {
+			if(typeof(service[p]) === 'function') {
+				methodNames.push(p);
+				this.onRequest(`service:${serviceType}:${version}:${p}`, service[p]);
```

Comment:
I'm assuming you mean `this` inside the service's function? When `service[p]` is eventually called by `sendRequestResponse` (`handler.apply(handler, w.args`), `this` will be the method itself.

So maybe it's worth mentioning that all the service methods have to essentially be static and not refer to `this` or other methods... unless I change how this works significantly.